echoregions/regions2d/regions2d_mask.py

```diff
@@ -0,0 +1,136 @@
+from typing import List
+
+import matplotlib
+import numpy as np
+import regionmask
+import xarray as xr
+from xarray import DataArray
+
+from .regions2d import Regions2D
+
+
+def regions2d_mask(
+    da_Sv: DataArray,
+    regions2d: Regions2D,
+    region_ids: List,
+    mask_var: str = None,
+    mask_labels=None,
+) -> DataArray:
```

Comment:
We want this to be a method `Regions2d.mask`, so the `regions2d` input argument would just be the `self`. Note the `replace_nan_depth` has a bug in doing `inplace=True` that needs to get fixed so that the nan_depth is replaced in `self.data` (I have this in my previous comment).